avoid default pagination limits

cboettig opened the pull request:
0.9 created a bug by introducing call to gh inside pb_info() (and thus all core functions using that method) which used left the default number of items (30) in place for assets returned. This patch makes sure we get all assets attached to the release.  

@karinorman can you check this when you get a chance? I think this should fix both issues you spotted in #23.  (previously you could not upload because piggyback didn't know it needed to delete the existing version, since it only saw the first 30 items due to this bug.  similarly, it would only download the first 30 items, all of which were in `col`).

cboettig committed:
avoid default pagination limits

0.9 created a bug by introducing call to gh inside pb_info() (and thus all core functions using that method) which used left the default number of items (30) in place for assets returned.

karinorman commented:
@cboettig I've successfully been able to upload single files with the dev version, but I'm still getting the gateway error for the whole database (on both my machine and the server). `install.packages` now installs version 0.0.9, so I haven't been able to test 0.0.8 to make sure it's not the machines/internet.

cboettig commented:
@karinorman Just to double-check: did you test with the `patch-bug` branch from GitHub? e.g.

```r
remotes::install_github("cboettig/piggyback@patch-bug")
```

cboettig committed:
also avoid non-file assets

(more precisely, omit objects returned by assets API endpoint as being in state "starter", which does not list on web end.  These may be folder objects).

cboettig committed:
bump version, NEWS

karinorman commented:
Whoops, nope, wasn't pulling from the branch.

Still no cigar though, now I'm getting the error: "Error in pb_upload_file(f, repo, tag, name, overwrite, use_timestamps,  : Unprocessable Entity (WebDAV; RFC 4918) (HTTP 422)."

cboettig commented:
Thanks for the checks, glad we're working now!
